lightning/src/ln/onion_utils.rs

```diff
@@ -1144,6 +1144,10 @@ where
 		.expect("Route we used spontaneously grew invalid keys in the middle of it?");
 	}
 
+	// In the best case, paths can be up to 27 hops. But attribution data can only be conveyed back to the sender from
+	// the first 20 hops. Determine the number of hops to be used for attribution data.
+	let attributable_hop_count = usize::min(path.hops.len(), MAX_HOPS);
```

Comment:
Still find myself [wishing](https://github.com/lightningdevkit/rust-lightning/pull/2256#discussion_r2006427003) that the caller didn't need to concern itself with this level of detail about attributable faliures, but don't see an obvious way to improve it.

Reply:
At the point of interpreting the failure message, there isn't much we can do about it. But one open question is whether we should limit pathfinding to 20 hops instead of 27 in pathfinding, so that we're sure every failure is attributable?